activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -746,10 +747,20 @@ def pk_and_sequence_for(table)
         end
       end
 
-      # Returns just a table's primary key
-      def primary_key(table)
-        pk_and_sequence = pk_and_sequence_for(table)
-        pk_and_sequence && pk_and_sequence.first
+      def primary_keys(table_name) # :nodoc:
+        return [] unless table_name.present?
```

Comment:
It's reasonable as an optimization since the query will always return an empty set. But why are we ever asking for the primary keys of a `nil` table? Seems like a case that we should treat as exceptional, or at least ignore, not one to optimize for.